crates/sqlexec/src/remote/client.rs

```diff
@@ -173,15 +157,24 @@ impl RemoteClient {
     }
 
     /// Connect to a proxy destination.
-    pub async fn connect_with_proxy_destination(dst: ProxyDestination) -> Result<Self> {
-        Self::connect_with_proxy_auth_params(dst.dst.to_string(), dst.params, dst.tls_conf).await
+    pub async fn connect_with_proxy_destination(
+        dst: ProxyDestination,
+        disable_tls: bool,
+    ) -> Result<Self> {
+        let mut dst: ProxyDestination = dst;
+        if !disable_tls {
+            dst.dst
+                .set_scheme("https")
+                .expect("failed to upgrade scheme from http to https");
+        }
+        Self::connect_with_proxy_auth_params(dst.dst.to_string(), dst.params, disable_tls).await
     }
 
     /// Connect to a destination with the provided authentication params.
     async fn connect_with_proxy_auth_params<'a>(
         dst: impl TryInto<Endpoint, Error = tonic::transport::Error>,
         params: ProxyAuthParams,
-        tls_conf: Option<TlsConfig>,
+        disable_tls: bool,
```

Comment:
I would prefer we change this to `enable_tls`. We have a few unneeded `!`s right now.

Reply:
I think there's decent cause to do the resolution of the cert during client creation not during connection (which would bring back tls config) and if the Option<> was populated it'd be enabled and if it weren't it would be disabled (right?)

Reply:
Fair, I am slightly worried about how that logic gets shared between Local and Python.

Is this something we can follow-up optimize (unless my worry is unfounded)?